examples/xpu/simple-fence/simple-fence.cpp

```diff
@@ -0,0 +1,170 @@
+        break;
+    case ISPCRT_DEVICE_TYPE_GPU:
+        os << "GPU";
+        break;
+    case ISPCRT_DEVICE_TYPE_CPU:
+        os << "CPU";
+        break;
+    default:
+        break;
+    }
+    return os;
+}
+
+struct Parameters {
+    float *vin;
+    float *vout;
+    int count;
+};
+
+void simple_CPU_validation(std::vector<float> vin, std::vector<float> &vgold, const unsigned int SIZE) {
+    for (unsigned int i = 0; i < SIZE; i++) {
+        float v = vin[i];
+        for (int i = 0; i < 100000; i++) {
+            if (v < 3.)
+                v = v * v;
+            else
+                v = std::sqrt(v);
+        }
+
+        vgold[i] = v;
+    }
+}
+
+#define EPSILON 0.01f
+bool validate_result(std::vector<float> vout, std::vector<float> vgold, const unsigned int SIZE) {
+    bool bValid = true;
+    for (unsigned int i = 0; i < SIZE; i++) {
+        float delta = (float)fabs(vgold[i] - vout[i]);
+        if (delta > EPSILON) {
+            std::cout << "Validation failed on i=" << i << ": vout[i] = " << vout[i] << ", but " << vgold[i]
+                      << " was expected\n";
+            bValid = false;
+        }
+    }
+    return bValid;
+}
+
+static int run(const ISPCRTDeviceType device_type, const unsigned int SIZE) {
+    std::vector<float> vin(SIZE);
+    std::vector<float> vout(SIZE);
+    std::vector<float> vgold(SIZE);
+
+    ispcrt::Device device(device_type);
+
+    // Setup input array
+    ispcrt::Array<float> vin_dev(device, vin);
+
+    // Setup output array
+    ispcrt::Array<float> vout_dev(device, vout);
+
+    // Setup parameters structure
+    Parameters p;
+
+    p.vin = vin_dev.devicePtr();
+    p.vout = vout_dev.devicePtr();
+    p.count = SIZE;
+
+    auto p_dev = ispcrt::Array<Parameters>(device, p);
+
+    // Create module and kernel to execute
+    ispcrt::Module module(device, "xe_simple-fence");
+    ispcrt::Kernel kernel(device, module, "simple_ispc");
+
+    // Create task queue and execute kernel
+    ispcrt::CommandQueue queue(device, 0);
+    ispcrt::CommandList cmds = queue.createCommandList();
+
+    std::generate(vin.begin(), vin.end(), [i = 0]() mutable { return i++; });
+
+    // Calculate gold result
+    simple_CPU_validation(vin, vgold, SIZE);
+
+    // ispcrt::Array objects which used as inputs for ISPC kernel should be
+    // explicitly copied to device from host
+    cmds.copyToDevice(p_dev);
+    cmds.copyToDevice(vin_dev);
+    cmds.barrier();
+    // Launch the kernel on the device using 1 thread
+    cmds.launch(kernel, p_dev, 1);
+    cmds.barrier();
+    // ispcrt::Array objects which used as outputs of ISPC kernel should be
+    // explicitly copied to host from device
+    cmds.copyToHost(vout_dev);
+
+    ispcrt::Fence fence = cmds.submit();
+    std::cout << "fence status " << fence.status() << std::endl;
+    while (!fence.status()) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(3));
+        std::cout << "Wait fence to be signaled" << std::endl;
```

Comment:
How many messages like this there will be typically? If it's going to be every 3ms and GPU is stall for a few ms for some reason, there might be a problem that there are too many of them.

Reply:
OK, I have increased up to 10ms. So now there are a couple of messages.
```bash
fence status 0
Wait fence to be signaled
Wait fence to be signaled
Wait finished
fence status 1
```